Src/Core/Primitives/ReferenceTypeAssertions.cs

```diff
@@ -146,6 +164,31 @@ public AndConstraint<TAssertions> BeOfType(Type expectedType, string because = "
         }
 
         /// <summary>
+        /// Asserts that the object is not of the specified type <paramref name="expectedType"/>.
+        /// </summary>
+        /// <param name="expectedType">
+        /// The type that the subject is not supposed to be of.
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion 
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> NotBeOfType( Type expectedType, string because = "", params object[] becauseArgs )
+        {
+            Execute.Assertion
+                .ForCondition( !ReferenceEquals( Subject, null ) )
```

Comment:
Well, somebody *could* pass a `null` into that method. And then you don't want to end up with an internal error.